src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -329,10 +328,10 @@ class DeepDiveLogParser extends JavaTokenParsers {
   ( implyHeadAtoms ^^ {
       InferenceRuleHead(FactorFunction.Imply, _)
     }
-  | "@semantics(linear)" ~> implyHeadAtoms ^^ {
+  | ("@semantics" ~ "(" ~ "\"linear\"" ~ ")") ~> implyHeadAtoms ^^ {
       InferenceRuleHead(FactorFunction.Linear, _)
     }
-  | "@semantics(ratio)" ~> implyHeadAtoms ^^ {
+  | ("@semantics" ~ "(" ~ "\"ratio\"" ~ ")") ~> implyHeadAtoms ^^ {
```

Comment:
Can we use `stringLiteralAsString` for the semantics value instead?  It seems `^?` can be pretty useful with partial functions.

```scala
  | ("@semantics" ~> "(" ~> stringLiteralAsString <~ ")" ^? ({
      case "linear" => FactorFunction.Linear
      case "ratio"  => FactorFunction.Ratio
    }, { "Unrecognized semantics: " + _ })) ~ implyHeadAtoms ^^ {
      case s ~ h => InferenceRuleHead(s, h)
    }
```